src/db/mock.rs

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Mock db implementation with methods stubbed to return default values.
+
+use futures::future;
+
+use super::*;
+
+#[derive(Debug)]
+pub struct MockDb;
+
+macro_rules! mock_db_method {
+    ($name:ident, $type:ident) => {
+        fn $name(&self, _params: params::$type) -> Box<Future<Item = results::$type, Error = DbError>> {
```

Comment:
It's already become pretty common for Box Future's with common Error types to type alias it away, for example in autopush-rs we have:
https://github.com/mozilla-services/autopush-rs/blob/master/src/errors.rs#L66

actix-web used the trait system all-out to not only type alias away the Box Future shenanigans, but also put a variety of traits around Futures in general for a nicer more ergonomic API internally and externally:
https://github.com/actix/actix-web/blob/master/src/handler.rs#L157

Using such a setup should get us better readability, without the boiler-plate that made macro's slightly more appealing in this case.